File: day-1/typescript/helpers/transformers.ts
import {InputPair} from "../types/input";
import {devMode} from "../constants/dev";

export const parseData = (data?: string): InputPair[] | undefined => {
    if (data) {
        const parsedData = data
            .trim()
            .split("\n")
            .map((line) => {
                const [first, second] = line.trim().split(/\s+/);
                return { valueOne: parseInt(first, 10), valueTwo: parseInt(second, 10) };
            });

        if (devMode) {
            console.log("parsed data: ", parsedData);
        }

        return parsedData;
    }
};

File: day-1/rust/src/helpers/transformers.rs
use crate::constants::dev::DEV_MODE;
use crate::types::types::InputPair;

pub fn parse_data(data: &Option<String>) -> Option<Vec<InputPair>> {
    if let Some(data_str) = data {
        let parsed_data: Vec<InputPair> = data_str
            .trim()
            .lines()
            .filter_map(|line| {
                let parts: Vec<&str> = line.trim().split_whitespace().collect();
                if parts.len() == 2 {
                    Some(InputPair {
                        value_one: parts[0].parse().ok()?,
                        value_two: parts[1].parse().ok()?,
                    })
                } else {
                    None
                }
            })
            .collect();

        if DEV_MODE {
            println!("Parsed Data: {:?}", parsed_data);
        }

        if parsed_data.is_empty() {
            None
        } else {
            Some(parsed_data)
        }
    } else {
        None
    }
}
